examples/cross_decomposition/plot_pcr_vs_pls.py

```diff
@@ -0,0 +1,116 @@
+
+##############################################################################
+# The data
+# --------
+#
+# We start by creating a simple dataset with two features. The target `y` is
+# strongly correlated with the second feature, which has a low variance. The
+# first feature has a higher variance but no predictive power on the target.
+
+import numpy as np
+import matplotlib.pyplot as plt
+
+rng = np.random.RandomState(0)
+n_samples = 500
+f0 = rng.normal(scale=10, size=n_samples)
+f1 = rng.normal(scale=1, size=n_samples)
+X = np.c_[f0, f1]
+y = f1 + rng.normal(scale=.1, size=n_samples)
+
+fig, axes = plt.subplots(1, 3, figsize=(10, 3))
+
+axes[0].scatter(f0, f1)
+axes[0].set(ylim=(-10, 10), xlim=(-20, 20), xlabel='f0', ylabel='f1',
+            title='training data')
+axes[1].scatter(f0, y)
+axes[1].set(xlabel='f0', ylabel='y', title='target vs f0')
+axes[2].scatter(f1, y)
+axes[2].set(xlabel='f1', ylabel='y', title='target vs f1')
+plt.tight_layout()
+plt.show()
+
+##############################################################################
+# Projection on one component and predictive power
+# ------------------------------------------------
+#
+# We now create two regressors: PCR and PLS, and for our illustration purposes
+# we set the number of components to 1. For both models, we plot the first
+# component against the target.
+#
+from sklearn.model_selection import train_test_split
+from sklearn.pipeline import make_pipeline
+from sklearn.linear_model import LinearRegression
+from sklearn.decomposition import PCA
+from sklearn.cross_decomposition import PLSRegression
+
+X_train, X_test, y_train, y_test = train_test_split(X, y, random_state=rng)
+
+pcr = make_pipeline(PCA(n_components=1), LinearRegression())
+pcr.fit(X_train, y_train)
+pca = pcr.named_steps['pca']  # retrieve the PCA step of the pipeline
+
+pls = PLSRegression(n_components=1)
+pls.fit(X_train, y_train)
+
+fig, axes = plt.subplots(1, 2, figsize=(10, 3))
+axes[0].scatter(pca.transform(X_test), y_test)
+axes[0].set(xlabel='first component', ylabel='y', title='PCR / PCA')
+axes[1].scatter(pls.transform(X_test), y_test)
+axes[1].set(xlabel='first component', ylabel='y', title='PLS')
+plt.tight_layout()
+plt.show()
+
+##############################################################################
+# As expected, the unsupervized PCA transformation of PCR has dropped the
+# second feature because it has a low variance, despite it being the most
+# predictive feature. This results in the first component having a low
+# predictive power on the target. On the other hand, the PLS regressor manages
+# to capture the effect of the second feature thanks to its use of target
+# information during the transformation: it can recogize that the second
+# feature should not be dropped.
+#
+# We also print the R-squared scores of both estimators, which further confirms
+# that PLS is a better alternative than PCR in this case:
+
+print(f"PCR r-squared {pcr.score(X_test, y_test):.3f}")
```

Comment:
Will readers gets confused about this being negative?

Reply:
I don't think so. Also I'm not sure there's much we can do about it?

Reply:
If we standardize `X`, the R^2 score for PRC gets positive, see https://github.com/scikit-learn/scikit-learn/pull/17151#discussion_r421656885.
On the other hand, one could add a comment that this estimator is worse than a `DummyRegressor`, and that a negative R^2 is very bad in general.

Reply:
If we're worried about a negative score, we should either stop exposing r2 as the score, or stop returning negative r2s. But we should be allowed to use our own default tools in the examples.

I'll add a comment indicating negative means worst than the mean